pkg/testhelper/testhelper.go

```diff
@@ -27,6 +31,90 @@ const (
 	ERROR
 )
 
+type ObjectOut struct {
+	ObjectType   string
+	ObjectFields map[string]string
+}
+
+type FailureReasonOut struct {
+	CompliantObjectsOut    []*ObjectOut
+	NonCompliantObjectsOut []*ObjectOut
+}
+
+const (
+	Namespace              = "Namespace"
+	PodName                = "PodName"
+	ContainerName          = "ContainerName"
+	ProcessID              = "ProcessID"
+	ProcessCommandLine     = "ProcessCommandLine"
+	SchedulingPolicy       = "SchedulingPolicy"
+	SchedulingPriority     = "SchedulingPriority"
+	ReasonForNonCompliance = "ReasonForNonCompliance"
+	ReasonForCompliance    = "ReasonForCompliance"
+	Category               = "Category"
+)
+
+const (
+	UndefinedType        = "Undefined Type"
+	CnfType              = "Cnf"
+	PodType              = "Pod"
+	ContainerType        = "Container"
+	ContainerProcessType = "ContainerProcess"
+)
+
+func NewProcessObjectOut(aContainer *ObjectOut, aPolicy, aPriority, aCommandLine string) (out *ObjectOut) {
+	out = aContainer
+	out.ObjectType = ContainerProcessType
+	out.ObjectFields[ProcessCommandLine] = aCommandLine
+	out.ObjectFields[SchedulingPolicy] = aPolicy
+	out.ObjectFields[SchedulingPriority] = aPriority
+	return out
+}
+
+func NewContainerObjectOut(aContainer *provider.Container, aReason string, isCompliant bool) (out *ObjectOut) {
+	return NewContainerObjectOutBase(aContainer.Namespace, aContainer.Podname, aContainer.Name, aReason, isCompliant)
+}
+
+func NewContainerObjectOutBase(aNamespace, aPodName, aContainerName, aReason string, isCompliant bool) (out *ObjectOut) {
```

Comment:
It looks unnecessary to have two versions of a function to create container objects.

Reply:
Might be a use-case for generics?

Reply:
fixed